### ISSUE TITLE
Phone accuracy of Mockingjay

### ISSUE THREAD

**Issue by jacky123465** [andi611/Self-Supervised-Speech-Pretraining-and-Representation-Learning#43]
Dear Author :
  Hello ! I am a Master from xi’an jiaotong university, thanks for your amazing open source project! I have executed the code of Mockingjay, but the model achieve 8%. phone accuracy in test-clean dataset and 70% in train dataset. 
In this pretrain process, I train on train-clean-360 dataset for 500k total training steps and use parameters in 'mockingjay_libri_melBase.yaml'. The target is mel-160 features. The specific parameters are as follows:
`transformer:
  input_dim: 160                                        # `int`, 39 for mfcc, 40 for fmllr, 80 for fbank, 160 for mel
  downsample_rate: 1                                    # stacked consecutive features vectors to reduce the length of input sequences by this factor.
  hidden_size: 768                                      # Size of the encoder layers and the pooler layer.
  num_hidden_layers: 3                                  # Number of hidden layers in the Transformer encoder.
  num_attention_heads: 12                               # Number of attention heads for each attention layer in the Transformer encoder.
  intermediate_size: 3072                               # The size of the "intermediate" (i.e., feed-forward) layer in the Transformer encoder.
  hidden_act: "gelu"                                    # The non-linear activation function (function or string) in the encoder and pooler. If string, "gelu", "relu" and "swish" are supported.
  hidden_dropout_prob: 0.1                              # The dropout probabilitiy for all fully connected layers in the embeddings, encoder, and pooler.
  attention_probs_dropout_prob: 0.1                     # The dropout ratio for the attention probabilities.
  initializer_range: 0.02                               # The sttdev of the truncated_normal_initializer for initializing all weight matrices.
  layer_norm_eps: "1e-12"                               # The epsilon used by LayerNorm.
  mask_proportion: 0.15                                 # mask this percentage of all spectrogram frames in each sequence at random during MAM training                        
  mask_consecutive_min: 7                               # mask this amount of consecutive frames
  mask_consecutive_max: 7                               # mask this amount of consecutive frames
  mask_allow_overlap: True                              # allow overlap masking
  mask_bucket_ratio: 1.2                                # only used when overlap is not allowed. sample a mask from each bucket in size of [sampled mask_consecutive * mask_bucket_ratio]
  mask_frequency: 0                                     # mask maximum this amount of frequency bands, set to 0 for no frequency mask
  noise_proportion: 0.0                                 # for this percentage of the time, Gaussian noise will be applied on all frames during MAM training, set to 0 for no noise
  prune_headids: None                                   # Usage: 0,1,2,12-15 will prune headids [0,1,2,12,13,14]. headids = layerid * head_num + headid_in_layer
  share_layer: False                                    # Share layer weights
  max_input_length: 0                                   # maximum input length (0 for no restriction)

optimizer: 
  type: 'adam'                                          # modes: ['adam', 'adamW', 'lamb']  
  learning_rate: "4e-4"                                 # Learning rate for opt. "4e-4" for 'data/libri_mel160_subword5000', "2e-4" for 'data/libri_fmllr_cmvn'
  loss_scale: 0                                         # Loss scale to improve fp16 numeric stability. Only used when apex is set to True. 0: dynamic loss scaling. positive power of 2: static loss scaling.
  warmup_proportion: 0.07                               # Proportion of training to perform linear rate warmup.
  gradient_accumulation_steps: 1                        # Number of updates steps to accumulate before performing a backward/update pass
  gradient_clipping: 1.0                                # Maximum gradient norm

dataloader:
  n_jobs: 12                                            # Subprocess used for torch Dataloader
  batch_size: 6                                         # training batch size
  dev_batch_size: 6                                     # used for dev/test splits
  max_timestep: 3000                                    # Max length for audio feature (0 for no restriction)
  
  # LIBRISEECH SETTINGS
  data_path: 'data/libri_mel160'                        # Source data path, 'data/libri_mel160_subword5000' or 'data/libri_fmllr_cmvn' for different preprocessing features
  target_path: 'data/libri_mel160'                      # Target data path for reconstruction to a different feature type, for example linear spectrograms
  phone_path: 'data/libri_phone'                        # phone boundary label data path for the phone classification task. set to 'data/libri_phone' or 'data/cpc_phone'
  train_set: ['train-clean-360']                        # ['train-clean-100', 'train-clean-360', 'train-other-500'] for pre-training. ['train-clean-360'] or ['train-clean-100'] for libri phone exp or cpc phone exp, respectively.
  dev_set: ['dev-clean']                                #
  test_set: ['test-clean']                              #
  train_proportion: 1.0                                 # Currently only effect the `phone classification task`, use this percent of `train_set` for downstream task training to demonstrate mockingjay generality

runner:
  # Training options
  apex: False                                           # Use APEX (see https://github.com/NVIDIA/apex for more details)
  total_steps: 500000                                   # total steps for training, a step is a batch of update
  log_step: 2500                                        # log training status every this amount of training steps
  save_step: 10000                                      # save model every this amount of training steps
  duo_feature: False                                    # Use different input / output features during training
  max_keep: 5                                           # maximum number of model ckpt to keep during training`

In this train process, I train on the all  train-clean-360 and use parameters. in 'downstream.yaml'. The specific parameters are as follows:
`dataloader:
  n_jobs: 6                                             # Subprocess used for torch Dataloader
  batch_size: 6                                         # training batch size
  dev_batch_size: 12                                    # used for dev/test splits
  max_timestep: 0                                       # Max length for audio feature (0 for no restriction)

  data_path: 'data/libri_mel160'                        # Source data path, 'data/libri_fmllr_cmvn', or 'data/libri_mfcc_cmvn', or 'data/libri_mel160_subword5000' for different preprocessing features
  phone_path: 'data/libri_phone'                        # phone boundary label data path for the phone classification task. set to 'data/libri_phone' or 'data/cpc_phone'
  libri_root: '/data/dataset/Libri/Libri/LibriSpeech/'      # only used when extract features on-the-fly
  train_set: ['train-clean-360']                        # ['train-clean-100', 'train-clean-360', 'train-other-500'] for pre-training. ['train-clean-360'] or ['train-clean-100'] for libri phone exp or cpc phone exp, respectively.
  dev_set: ['test-clean']                               #
  test_set: ['test-clean']                              #
  train_proportion: 1.0                                 # Currently only effect the `phone classification task`, use this percent of `train_set` for downstream task training to demonstrate mockingjay generality

runner:
  learning_rate: '4e-3'                                 # Learning rate for opt: ['4e-3' for fine-tune, '4e-3' for regualr downstream task training]
  warmup_proportion: 0.07                               # Proportion of training to perform linear rate warmup.
  gradient_clipping: 1.0                                # Maximum gradient norm
  total_steps: 500000                                   # total steps for training, a step is a batch of update
  log_step: 2000                                        # log training status every this amount of training steps
  save_step: 2000                                       # save model every this amount of training steps
  dev_step: 5000                                        # evaluate every this amount of training steps
  evaluation: 'test'                                    # can be 'dev' or 'test', show inference results and saves the best model
  max_keep: 2                                           # maximum number of model ckpt to keep during training

model:                                                  # downstream model config, each task can have different model setting
  
  phone_linear:
    hidden_size: 0                                      # when linear: True, the hidden_size is ignored
    drop: 0.0                                           # The dropout ratio, not used when `linear` is set to `True`.
    linear: True                                        # whether to make the classifier linear
    layers: 1                                           # number of layers in the classifier, set to 2 for 1 hidden layer
    concat: 1                                           # `int`, must be an odd number. Concatenation of this amount of windows to match the average size of a phoneme. Set to 1 for no concatenation, set to 9 to concat 4 previous and 4 future frames.

  phone_1hidden:
    hidden_size: 768                                    # hidden size of classifier
    drop: 0.0                                           # The dropout ratio, not used when `linear` is set to `True`.
    linear: False                                       # whether to make the classifier linear
    layers: 2                                           # number of layers in the classifier, set to 2 for 1 hidden layer
    concat: 1                                           # `int`, must be an odd number. Concatenation of this amount of windows to match the average size of a phoneme. Set to 1 for no concatenation, set to 9 to concat 4 previous and 4 future frames.
  
  phone_concat:
    hidden_size: 0                                      # when linear: True, the hidden_size is ignored
    drop: 0.0                                           # The dropout ratio, not used when `linear` is set to `True`.
    linear: True                                        # whether to make the classifier linear
    layers: 1                                           # number of layers in the classifier, set to 2 for 1 hidden layer
    concat: 9                                           # `int`, must be an odd number. Concatenation of this amount of windows to match the average size of a phoneme. Set to 1 for no concatenation, set to 9 to concat 4 previous and 4 future frames.

  speaker_frame:
    hidden_size: 0                                      # when linear: True, the hidden_size is ignored
    drop: 0.0                                           # The dropout ratio, not used when `linear` is set to `True`.
    linear: True                                        # whether to make the classifier linear
    layers: 1                                           # number of layers in the classifier, set to 2 for 1 hidden layer
    concat: 1                                           # `int`, must be an odd number. Concatenation of this amount of windows to match the average size of a phoneme. Set to 1 for no concatenation, set to 9 to concat 4 previous and 4 future frames.

  speaker_utterance:
    hidden_size: 0                                      # when linear: True, the hidden_size is ignored
    drop: 0.0                                           # The dropout ratio, not used when `linear` is set to `True`.
    linear: True                                        # whether to make the classifier linear
    layers: 1                                           # number of layers in the classifier, set to 2 for 1 hidden layer
    concat: 1                                           # `int`, must be an odd number. Concatenation of this amount of windows to match the average size of a phoneme. Set to 1 for no concatenation, set to 9 to concat 4 previous and 4 future frames.`

I don't know what caused this result. Would you like to help me? Thanks you very much !
